src/datatypes/src/schema/column_schema.rs

```diff
@@ -262,6 +262,14 @@ impl ColumnSchema {
         );
         Ok(self)
     }
+
+    pub fn set_fulltext_options(&mut self, options: FulltextOptions) -> Result<()> {
+        self.metadata.insert(
+            FULLTEXT_KEY.to_string(),
+            serde_json::to_string(&options).context(error::SerializeSnafu)?,
+        );
+        Ok(())
+    }
```

Comment:
Prefer to reduce the duplicated code

Option 1: refactor the `with_fulltext_options`  method
```rust
    pub fn with_fulltext_options(mut self, options: FulltextOptions) -> Result<Self> {
        self. set_fulltext_options(options)?;
        Ok(self)
    }
```

Option 2: use the `with_fulltext_options ` method
```rust
column_meta.column_schema = column_meta.column_schema.with_fulltext_options(fulltext)?;
```